hooks/pre/31-mongo_storage_engine.rb

```diff
@@ -57,7 +57,7 @@ def migration
 
   # Update Hiera to wiredTiger for installer run
   logger.info 'Changing custom Hiera to use wiredTiger for installer Puppet run.'
-  Kafo::Helpers.execute("sed -i -e 's/run the installer with --upgrade-mongo-storage to upgrade to wiredTiger./Do not remove'/g #{hiera_file}")
+  Kafo::Helpers.execute("sed -i -e 's/Added by foreman-installer during upgrade, run the installer with --upgrade-mongo-storage to upgrade to WiredTiger./Do not remove'/g #{hiera_file}")
```

Comment:
Should that be `--upgrade-mongo-storage-engine` ?

Reply:
@ehelms this is removing the line we did during the upgrade, this is a 6.4 hiera file that has been upgraded from 6.3 which has the old wording. So we want --upgrade-mongo-storage

```
[root@satellite1 ~]# cat /etc/foreman-installer/custom-hiera.yaml 
# Added by foreman-installer during upgrade, run the installer with --upgrade-mongo-storage to upgrade to WiredTiger.
mongodb::server::storage_engine: 'mmapv1'
```

Reply:
The `sed` and your output are confusing me. As I still see the wrong installer argument being put into the hiera file? As in, I don't see it being removed based on your `cat`.

Reply:
@ehelms here is a better way to look at it, hopefully, easier to follow. 
```
Before engine upgrade run:

# Added by foreman-installer during upgrade, run the installer with --upgrade-mongo-storage to upgrade to WiredTiger.
mongodb::server::storage_engine: 'mmapv1'

Command being run:

[ INFO 2018-12-12T11:17:45 verbose] Changing custom Hiera to use wiredTiger for installer Puppet run.
sed -i -e 's/Added by foreman-installer during upgrade, run the installer with --upgrade-mongo-storage to upgrade to WiredTiger./Do not remove'/g /etc/foreman-installer/custom-hiera.yaml finished successfully!
sed -i -e 's/mmapv1/wiredTiger/g' /etc/foreman-installer/custom-hiera.yaml finished successfully!

File after engine upgrade:

# Do not remove
mongodb::server::storage_engine: 'wiredTiger'
```

Reply:
Before it would keep this line in there:
```
# Added by foreman-installer during upgrade, run the installer with --upgrade-mongo-storage to upgrade to WiredTiger.
```
The updated sed makes the hiera file have the correct wording.